### PR TITLE
chore: Updating tests and adding documentation for error handling in …

### PR DESCRIPTION
**Pull request by vanshu86**
#minor  update in Call Transfer

### Purpose
chore: Updating tests and adding documentation for error handling in Call Transfer. 

### Changes
As discussed, we have the minimal set of tests. For error handling we have updated documentation.

### Tests
added tests

### Feature Plan
n/a

### CODE REVIEW

File: packages/Telephony/Readme.md [hunk 1]
```diff
@@ -13,18 +13,17 @@ Like any other channel, Telephony channel allows you to transfer call to an agen
...
 
 #### Usage
-* Phone Number is not empty and in the E.164 format. 
+* Phone Number should not empty and in the E.164 format. 
 * The call transfer action is only valid when called in a conversation on the Telephony channel. The action can be considered a No-op for all other channels.
 
 #### Dialog Flow
 * Once the call transfer is completed, the bot is removed from the current conversation and control is transferred to the extenal phone nunber.
-* [Open] Do we get back a handoff status on success? Should we wait for the status in our package?
+* We don't get any handoff status on success.
```
[the remark this answers is not in the document]
Comment (vanshu86) [reply packages/Telephony/Readme.md]: for sure!

File: packages/Telephony/Readme.md [hunk 2]
```diff
@@ -13,18 +13,17 @@ Like any other channel, Telephony channel allows you to transfer call to an agen
 * PhoneNumber
 
 #### Usage
-* Phone Number is not empty and in the E.164 format. 
+* Phone Number should not empty and in the E.164 format.
```
[the remark this answers is not in the document]
Comment (vanshu86) [reply packages/Telephony/Readme.md]: yep :)

File: packages/Telephony/Readme.md [hunk 3]
```diff
@@ -13,18 +13,17 @@ Like any other channel, Telephony channel allows you to transfer call to an agen
 * PhoneNumber
 
 #### Usage
-* Phone Number is not empty and in the E.164 format. 
+* Phone Number should not be empty and should be in the E.164 format. 
 * The call transfer action is only valid when called in a conversation on the Telephony channel. The action can be considered a No-op for all other channels.
 
 #### Dialog Flow
 * Once the call transfer is completed, the bot is removed from the current conversation and control is transferred to the extenal phone nunber.
```
Comment (p-nagpal) [inline packages/Telephony/Readme.md:20]: typo: 
extenal --> external
nunber --> number

File: packages/Telephony/Readme.md [hunk 4]
```diff
@@ -13,18 +13,17 @@ Like any other channel, Telephony channel allows you to transfer call to an agen
...
-* [Open] Do we get back a handoff status on success? Should we wait for the status in our package?
+* The bot will not get any handoff status on success.
 * Any actions specified after call transfer will not be executed. Treat it like a call end.
```
Comment (p-nagpal) [inline packages/Telephony/Readme.md:22]: Will the subsequent actions be executed after a failed call transfer?

File: packages/Telephony/Schemas/Microsoft.Telephony.CallTransfer.uischema [hunk 5]
```diff
@@ -1,8 +1,8 @@
 ﻿{
   "$schema": "https://schemas.botframework.com/schemas/ui/v1.0/ui.schema",
   "form": {
-    "label": "Transfer a call",
-    "subtitle": "Call Transfer",
+    "label": "Call Transfer",
+    "subtitle": "Transfers call to phone number specified (in E.164 format such as +1425123456)",
```
Comment (p-nagpal) [inline packages/Telephony/Schemas/Microsoft.Telephony.CallTransfer.uischema:5]: grammar: Transfers call to _the_ phone number specified

File: tests/packages/Microsoft.Bot.Components.Telephony.Tests/CallTransferTests.cs [hunk 6]
```diff
@@ -20,6 +20,12 @@ public CallTransferTests(ResourceExplorerFixture resourceExplorerFixture) : base
         public async Task CallTransfer_HappyPath()
         {
             await TestUtils.RunTestScript(_resourceExplorerFixture.ResourceExplorer, adapterChannel: Channels.Telephony);
+        }
+		
+		[Fact]
```
Comment (p-nagpal) [inline tests/packages/Microsoft.Bot.Components.Telephony.Tests/CallTransferTests.cs:25]: Fix tab/spacing

File: tests/packages/Microsoft.Bot.Components.Telephony.Tests/Integration/CallTransferTests/CallTransfer_BaseScenario.test.dialog [hunk 7]
```diff
@@ -5,12 +5,16 @@
     "triggers": [
         {
             "$kind": "Microsoft.OnBeginDialog",
-            "actions": [
-                {
-                    "$kind": "Microsoft.Telephony.CallTransfer",
-                    "phoneNumber": "+1555443443"
-                }
-            ]
+          "actions": [
```
Comment (p-nagpal) [inline tests/packages/Microsoft.Bot.Components.Telephony.Tests/Integration/CallTransferTests/CallTransfer_BaseScenario.test.dialog:8]: Minor: spacing in all test scripts seems off.

File: tests/packages/Microsoft.Bot.Components.Telephony.Tests/Integration/CallTransferTests/CallTransfer_HappyPath.test.dialog [hunk 8]
```diff
@@ -2,17 +2,21 @@
...
+    },
+    {
+      "$kind": "Microsoft.Test.AssertReplyActivity",
+      "assertions": [
+        "type == 'event'",
+        "value.TargetPhoneNumber == '+15554434432'"
+      ]
+    },
+    {
+      "$kind": "Microsoft.Test.AssertReply",
```
Comment (p-nagpal) [inline tests/packages/Microsoft.Bot.Components.Telephony.Tests/Integration/CallTransferTests/CallTransfer_HappyPath.test.dialog:18]: I know we went back and forth on this but this activity seems to conflict with the spec. We say no activity / action executes after call transfer but I can see that its hard to simulate in a mock. Do you know if we can add a comment to explain that this?

**Review by p-nagpal — APPROVED** [review]
Looks good. Minor feedback on style/docs

File: packages/Telephony/Readme.md (hunk 4 above)
Comment (vanshu86) [reply packages/Telephony/Readme.md:22]: yes, will will add that in failure flow too.

File: packages/Telephony/Readme.md [hunk 9]
```diff
@@ -13,18 +13,17 @@ Like any other channel, Telephony channel allows you to transfer call to an agen
...
 
 #### Usage
-* Phone Number is not empty and in the E.164 format. 
+* Phone Number should not be empty and should be in the E.164 format. 
 * The call transfer action is only valid when called in a conversation on the Telephony channel. The action can be considered a No-op for all other channels.
 
 #### Dialog Flow
 * Once the call transfer is completed, the bot is removed from the current conversation and control is transferred to the extenal phone nunber.
-* [Open] Do we get back a handoff status on success? Should we wait for the status in our package?
+* The bot will not get any handoff status on success.
```
Comment (p-nagpal) [inline packages/Telephony/Readme.md:21]: The current behavior is 
1. Handoff Success : gets End of Conversation and the call ends
2. Handoff failure: gets a handoff status with a failed message.

Comment (vanshu86) [reply packages/Telephony/Readme.md:21]: yes added in failure flow

**Review by carlosscastro — APPROVED** [review]

**Pull request merged**